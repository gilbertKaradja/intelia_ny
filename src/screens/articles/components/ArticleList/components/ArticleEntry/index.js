import React from 'react';

import FadeImage from '../../../../../../components/FadeImage/index.js';

import icon from '../../../../../../assets/icons/right-chevron.svg';

const ArticleEntry = (props) => {



    let imageLocation;

    try {
        props.media[0]['media-metadata'].map(item => {
            if (item.format === 'Standard Thumbnail') {
                imageLocation = item.url;
            }

            return item.url;
        });
    } catch (e) {
        imageLocation = '';
    }


    return (
        <div className={props.selected ? 'ArticleEntry selected' : 'ArticleEntry'} onClick={props.onClickHandler}>

            <div className="thumbnail_wrapper">
                <FadeImage src={imageLocation} name='thumbnail' />
            </div>

            <div className="details_wrapper">

                <div className="title_author">
                    <div className="title block-with-text">{props.title}</div>
                    <div className="author">{props.byline}</div>

                    <div className="date">
                        <label>{props.published_date}</label>
                    </div>

                </div>



            </div>

            <div className="button_wrapper">
                <img src={icon} alt='caret_right' />
            </div>

        </div>
    );
}

export default ArticleEntry;