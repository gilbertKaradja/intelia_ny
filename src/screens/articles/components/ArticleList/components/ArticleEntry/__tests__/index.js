import React from 'react';
import { shallow, mount } from 'enzyme';

import ArticleEntry from '../index.js';

let sampleData = {
    "url": "https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html",

    "section": "The Upshot",
    "byline": "By JOSH KATZ",

    "title": "The British-Irish Dialect Quiz",
    "abstract": "What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map.",
    "published_date": "2019-02-15",
    "source": "The New York Times",
    "id": 100000006354694,

    "media": [
        {
            "media-metadata": [
                {
                    "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-thumbStandard-v3.jpg",
                    "format": "Standard Thumbnail",
                    "height": 75,
                    "width": 75
                },
                {
                    "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-mediumThreeByTwo440-v3.jpg",
                    "format": "mediumThreeByTwo440",
                    "height": 293,
                    "width": 440
                }
            ]
        }
    ]
};

let sampleData2 = {
    "url": "https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html",

    "section": "The Upshot",
    "byline": "By JOSH KATZ",

    "title": "The British-Irish Dialect Quiz",
    "abstract": "What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map.",
    "published_date": "2019-02-15",
    "source": "The New York Times",
    "id": 100000006354694,

    "media": ''
};



describe('DetailsScreen component', () => {

    it('should render without issues', () => {

        //renders initial dropdown properly
        const component = shallow(
            <ArticleEntry
                {...sampleData}
                selected={true}
            />
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });


    it('should render properly even if an image is not available', () => {

        //renders initial dropdown properly
        const component = shallow(
            <ArticleEntry
                {...sampleData2}
                selected={false}
            />
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });



    it('should call the back handler function when back button is clicked/pressed', () => {

        const onClickHandler = jest.fn(() => sampleData);

        const component = mount(
            <ArticleEntry
                articleData={sampleData}
                {...sampleData}
                onClickHandler={onClickHandler}
            />
        );

        expect(component).toMatchSnapshot();

        component.find('.ArticleEntry').simulate('click');

        expect(onClickHandler.mock.calls.length).toBe(1);
        expect(onClickHandler.mock.results[0].value.id).toBe(sampleData.id);

        component.unmount();
    });
});