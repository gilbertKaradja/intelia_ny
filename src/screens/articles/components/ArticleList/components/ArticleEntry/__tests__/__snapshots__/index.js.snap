// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen component should call the back handler function when back button is clicked/pressed 1`] = `
<ArticleEntry
  abstract="What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map."
  articleData={
    Object {
      "abstract": "What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map.",
      "byline": "By JOSH KATZ",
      "id": 100000006354694,
      "media": Array [
        Object {
          "media-metadata": Array [
            Object {
              "format": "Standard Thumbnail",
              "height": 75,
              "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-thumbStandard-v3.jpg",
              "width": 75,
            },
            Object {
              "format": "mediumThreeByTwo440",
              "height": 293,
              "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-mediumThreeByTwo440-v3.jpg",
              "width": 440,
            },
          ],
        },
      ],
      "published_date": "2019-02-15",
      "section": "The Upshot",
      "source": "The New York Times",
      "title": "The British-Irish Dialect Quiz",
      "url": "https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html",
    }
  }
  byline="By JOSH KATZ"
  id={100000006354694}
  media={
    Array [
      Object {
        "media-metadata": Array [
          Object {
            "format": "Standard Thumbnail",
            "height": 75,
            "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-thumbStandard-v3.jpg",
            "width": 75,
          },
          Object {
            "format": "mediumThreeByTwo440",
            "height": 293,
            "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-mediumThreeByTwo440-v3.jpg",
            "width": 440,
          },
        ],
      },
    ]
  }
  onClickHandler={[MockFunction]}
  published_date="2019-02-15"
  section="The Upshot"
  source="The New York Times"
  title="The British-Irish Dialect Quiz"
  url="https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html"
>
  <div
    className="ArticleEntry"
    onClick={[MockFunction]}
  >
    <div
      className="thumbnail_wrapper"
    >
      <FadeImage
        name="thumbnail"
        src="https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-thumbStandard-v3.jpg"
      >
        <div
          className="FadeImage"
        >
          <img
            alt="thumbnail"
            onLoad={[Function]}
            src="https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-thumbStandard-v3.jpg"
          />
          <div
            className="placeholder"
          />
        </div>
      </FadeImage>
    </div>
    <div
      className="details_wrapper"
    >
      <div
        className="title_author"
      >
        <div
          className="title block-with-text"
        >
          The British-Irish Dialect Quiz
        </div>
        <div
          className="author"
        >
          By JOSH KATZ
        </div>
        <div
          className="date"
        >
          <label>
            2019-02-15
          </label>
        </div>
      </div>
    </div>
    <div
      className="button_wrapper"
    >
      <ReactComponent>
        <svg>
          right-chevron.svg
        </svg>
      </ReactComponent>
    </div>
  </div>
</ArticleEntry>
`;

exports[`DetailsScreen component should render properly even if an image is not available 1`] = `
<div
  className="ArticleEntry"
>
  <div
    className="thumbnail_wrapper"
  >
    <FadeImage
      name="thumbnail"
      src=""
    />
  </div>
  <div
    className="details_wrapper"
  >
    <div
      className="title_author"
    >
      <div
        className="title block-with-text"
      >
        The British-Irish Dialect Quiz
      </div>
      <div
        className="author"
      >
        By JOSH KATZ
      </div>
      <div
        className="date"
      >
        <label>
          2019-02-15
        </label>
      </div>
    </div>
  </div>
  <div
    className="button_wrapper"
  >
    <ReactComponent />
  </div>
</div>
`;

exports[`DetailsScreen component should render without issues 1`] = `
<div
  className="ArticleEntry selected"
>
  <div
    className="thumbnail_wrapper"
  >
    <FadeImage
      name="thumbnail"
      src="https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-thumbStandard-v3.jpg"
    />
  </div>
  <div
    className="details_wrapper"
  >
    <div
      className="title_author"
    >
      <div
        className="title block-with-text"
      >
        The British-Irish Dialect Quiz
      </div>
      <div
        className="author"
      >
        By JOSH KATZ
      </div>
      <div
        className="date"
      >
        <label>
          2019-02-15
        </label>
      </div>
    </div>
  </div>
  <div
    className="button_wrapper"
  >
    <ReactComponent />
  </div>
</div>
`;
